test/e2e/tests/test_sagemaker_endpoint_autoscaling.py

```diff
@@ -0,0 +1,201 @@
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["SCALABLETARGET_NAME"] = target_resource_name
+    replacements["RESOURCE_ID"] = resource_id
+
+    target_reference, _, target_resource = create_applicationautoscaling_resource(
+        resource_plural=TARGET_RESOURCE_PLURAL,
+        resource_name=target_resource_name,
+        spec_file="sagemaker_endpoint_autoscaling_target",
+        replacements=replacements,
+    )
+
+    assert target_resource is not None
+
+    yield (resource_id, target_reference)
+
+    if k8s.get_resource_exists(target_reference):
+        _, deleted = k8s.delete_custom_resource(target_reference)
+        assert deleted
+
+
+@pytest.fixture(scope="module")
+def generate_sagemaker_policy(generate_sagemaker_target):
+    resource_id, target_reference = generate_sagemaker_target
+    policy_resource_name = random_suffix_name("sagemaker-scaling-policy", 32)
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["SCALINGPOLICY_NAME"] = policy_resource_name
+    replacements["RESOURCE_ID"] = resource_id
+
+    policy_reference, _, policy_resource = create_applicationautoscaling_resource(
+        resource_plural=POLICY_RESOURCE_PLURAL,
+        resource_name=policy_resource_name,
+        spec_file="sagemaker_endpoint_autoscaling_policy",
+        replacements=replacements,
+    )
+
+    assert policy_resource is not None
+    assert k8s.get_resource_arn(policy_resource) is not None
+
+    yield (resource_id, target_reference, policy_resource, policy_reference)
+
+    if k8s.get_resource_exists(policy_reference):
+        _, deleted = k8s.delete_custom_resource(policy_reference)
+        assert deleted
+
+
+@service_marker
+@pytest.mark.canary
+class TestSageMakerEndpointAutoscaling:
+    def get_sagemaker_scalable_target_description(
+        self, applicationautoscaling_client, resource_id: str, expectedTargets: int
+    ):
+        try:
+            targets = applicationautoscaling_client.describe_scalable_targets(
+                ServiceNamespace="sagemaker",
+                ResourceIds=[resource_id],
+            )
+
+            assert len(targets["ScalableTargets"]) == expectedTargets
+            return targets["ScalableTargets"]
+        except BaseException:
+            logging.error(
+                f"ApplicationAutoscaling could not find a scalableTarget for the resource {resource_id}"
+            )
+            return None
+
+    def get_sagemaker_scaling_policy_description(
+        self, applicationautoscaling_client, resource_id: str, expectedPolicies: int
+    ):
+        try:
+            policies = applicationautoscaling_client.describe_scaling_policies(
+                ServiceNamespace="sagemaker",
+                ResourceId=resource_id,
+            )
+            assert len(policies["ScalingPolicies"]) == expectedPolicies
+            return policies["ScalingPolicies"]
+        except BaseException:
```

Comment:
change BaseException to `botocore.exceptions.ClientError` and print the error in catch block

this was a problem in sagemaker canaries because error was not being printed

Reply:
done, good point!